service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java

```diff
@@ -147,15 +153,81 @@
 
   protected ThreadLocal<ServerContext> currentServerContext;
 
+  /**
+   * A context container for information related to client connections to the
+   * Thrift server.
+   */
   static class ThriftCLIServerContext implements ServerContext {
-    private SessionHandle sessionHandle = null;
 
-    public void setSessionHandle(SessionHandle sessionHandle) {
-      this.sessionHandle = sessionHandle;
+    private final AtomicInteger messagesProcessedCounter;
```

Comment:
Maybe a normal integer counter is ok here,  there should be no thread safe problem...

Reply:
Done.  Thanks.